ImageFilterEditor/ImageFilterEditor/View Controller/ImagePostViewController.swift

```diff
@@ -0,0 +1,333 @@
+    
+    var originalImage: UIImage? {
+        didSet {
+            // resize the scaledImage and set it
+            guard let originalImage = originalImage else { return }
+            // Height and width
+            var scaledSize = imageView.bounds.size
+            let scale = UIScreen.main.scale  // 1x, 2x, or 3x
+            scaledSize = CGSize(width: scaledSize.width * scale, height: scaledSize.height * scale)
+            print("scaled size: \(scaledSize)")
+            
+            scaledImage = originalImage.imageByScaling(toSize: scaledSize)
+        }
+    }
+    var scaledImage: UIImage? {
+        didSet {
+            updateViews()
+        }
+    }
+    
+    // MARK: - View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        secondAdjustmentSlider.isHidden = true
+        adjustmentSlider.isHidden = true
+        nameLabel.isHidden = true
+        saveButton.isHidden = true
+    }
+    
+    // MARK: - IBActions
+    @IBAction func selectPhotoButtonTapped(_ sender: Any) {
+        presentImagePickerController()
+    }
+    
+    @IBAction func adjustmentSliderChanged(_ sender: Any) {
+        updateViews(withAdjustment: true)
+    }
+    
+    @IBAction func secondAdjustmentSliderChanged(_ sender: Any) {
+        updateViews(withAdjustment: true)
+    }
+    
+    @IBAction func saveButtonTapped(_ sender: Any) {
+        scaledImage = imageView.image
+    }
+    
+    // MARK: - Private Methods
+    private func presentImagePickerController() {
+        guard UIImagePickerController.isSourceTypeAvailable(.photoLibrary) else {
+            print("Error: The photo library is not available")
+            return
+        }
+        let imagePicker = UIImagePickerController()
+        imagePicker.sourceType = .photoLibrary
+        imagePicker.delegate = self
+        present(imagePicker, animated: true, completion: nil)
+    }
+    
+    private func updateViews(withAdjustment: Bool = false) {
+        if let scaledImage = scaledImage {
+            
+            if withAdjustment {
+                if filterType.rawValue == 0 {
+                    imageView.image = adjustExposure(scaledImage)
+                } else if filterType.rawValue == 1 {
+                    imageView.image = adjustVibrance(scaledImage)
+                } else if filterType.rawValue == 2 {
+                    imageView.image = adjustVignette(scaledImage)
+                } else if filterType.rawValue == 3 {
+                    imageView.image = adjustSepia(scaledImage)
+                } else if filterType.rawValue == 4 {
+                    imageView.image = adjustMotionBlur(scaledImage)
+                }
+            } else {
+                imageView.image = scaledImage
+            }
+            
+        } else {
+            imageView.image = nil
+        }
+    }
```

Comment:
Using a guard let can make the code below more readable because it's not indented as much, and the error condition is satisfied at the top

```swift
private func updateViews(withAdjustment: Bool = false) {
    guard let scaledImage = scaledImage else {
        imageView.image = nil
        return
    }
    
    if withAdjustment {
        if filterType.rawValue == 0 {
            imageView.image = adjustExposure(scaledImage)
        } else if filterType.rawValue == 1 {
            imageView.image = adjustVibrance(scaledImage)
        } else if filterType.rawValue == 2 {
            imageView.image = adjustVignette(scaledImage)
        } else if filterType.rawValue == 3 {
            imageView.image = adjustSepia(scaledImage)
        } else if filterType.rawValue == 4 {
            imageView.image = adjustMotionBlur(scaledImage)
        }
    } else {
        imageView.image = scaledImage
    }
}
```